source/common/tls/server_context_impl.cc

```diff
@@ -95,28 +95,16 @@ ServerContextImpl::ServerContextImpl(Stats::Scope& scope,
                                      Ssl::ContextAdditionalInitFunc additional_init,
                                      absl::Status& creation_status)
     : ContextImpl(scope, config, factory_context, additional_init, creation_status),
+      tls_certificate_selector_(config.createTlsCertificateSelector()(config, *this)),
```

Comment:
I think the call to `config.createTlsCertificateSelector()(config, *this)` should be after the check on line 101 for `!creation_status.ok` so the cert selector doesn't need to handle those error conditions.